### DISCUSSION TITLE
lwIP adapter layer

### DISCUSSION THREAD

**Discussion by chrismerck** [openthread/openthread#8753] [Q&A] [answered]
Does anyone know of an existing project to implement the OpenThread API against the lwIP raw API?

Motivation: Such a project would enable portability of OpenThread applications to WiFi- and Ethernet-based embedded systems.

The idea is that application components for a small Thread SoC can be written directly against the ot C++ API for smallest code-size and best performance. Then those components can be shared with an lwIP-based embedded system, or even a POSIX system if a sockets layer were added. 

To be clear, I'm not asking how to make OpenThread and lwIP co-exist. I'm not asking how to interface with a Thread network or even talk the Thread protocol from a Wi-Fi device. Rather, I'm trying to make OpenThread applications portable to non-Thread embedded products.

🙏

**Comment by jwhui** [comment, marked as answer]
Take a look at https://github.com/openthread/ot-rtos

It hasn't been maintained for some time, but should give a good starting point.
